modules/file/file_test.go

```diff
@@ -488,18 +472,12 @@ func TestMismatch(t *testing.T) {
 				Paths:    []string{basedir},
 				MD5:      []string{TESTDATA[2].md5},
 				SHA1:     []string{TESTDATA[2].sha1},
-				SHA256:   []string{TESTDATA[2].sha256},
-				SHA384:   []string{TESTDATA[2].sha384},
-				SHA512:   []string{TESTDATA[2].sha512},
-				SHA3_224: []string{TESTDATA[2].sha3_224},
-				SHA3_256: []string{TESTDATA[2].sha3_256},
-				SHA3_384: []string{TESTDATA[2].sha3_384},
-				SHA3_512: []string{TESTDATA[2].sha3_512},
+				SHA2:     []string{TESTDATA[2].sha2},
+				SHA3:     []string{TESTDATA[2].sha3},
 				Options: options{
 					MaxDepth: 1,
 					MatchAll: true,
-					Mismatch: []string{`md5`, `sha1`, `sha256`, `sha384`, `sha512`,
-						`sha3_224`, `sha3_256`, `sha3_384`, `sha3_512`},
+					Mismatch: []string{`md5`, `sha1`, `sha2`, `sha3`},
```

Comment:
This test that is failing: `file_test.go:490: want files that don't match the hashes of testfile2, should find testfile0 & 1`
The parameters used by the test are:
```json
{
	"class": "parameters",
	"parameters": {
		"searches": {
			"s1": {
				"paths": ["/tmp/migfiletest08-45-21.93044607"],
				"md5": ["8d3a7afb7e59693b383d52396243a5b8"],
				"sha1": ["d82bc1145d471714b056940b268032f9ab0df2ae"],
				"sha2": ["3b495fae5bae9751ea4706c29e992002ba277bce30bd83a827b01ba977eabc2f"],
				"sha3": ["fdb23afa808c265284c3199013e4ded9704eebf54ffdc1f016dacc12"],
				"options": {
					"maxdepth": 1,
					"matchall": true,
					"macroal": false,
					"mismatch": ["md5", "sha1", "sha2", "sha3"],
					"matchlimit": 0
				}
			}
		}
	}
}
```
And it returns:
```json
{
	"foundanything": true,
	"success": true,
	"elements": {
		"s1": [{
			"file": "",
			"search": {
				"paths": ["/tmp/migfiletest08-45-21.93044607"],
				"md5": ["8d3a7afb7e59693b383d52396243a5b8"],
				"sha1": ["d82bc1145d471714b056940b268032f9ab0df2ae"],
				"sha2": ["3b495fae5bae9751ea4706c29e992002ba277bce30bd83a827b01ba977eabc2f"],
				"sha3": ["fdb23afa808c265284c3199013e4ded9704eebf54ffdc1f016dacc12"],
				"options": {
					"maxdepth": 0,
					"matchall": true,
					"macroal": false,
					"mismatch": ["md5", "sha1", "sha2", "sha3"],
					"matchlimit": 0
				}
			},
			"fileinfo": {
				"size": 0,
				"mode": "",
				"lastmodified": ""
			}
		}]
	},
	"statistics": {
		"filescount": 243,
		"openfailed": 0,
		"totalhits": 73,
		"exectime": "1.03271ms"
	},
	"errors": null
}
```
Where `"file": "",` indicates that nothing was found, but the test expected to find testfile0 and testfile1 because it `mismatch`es on the hashes of testfile2, and thus should find all other files.
I need to dig through your patch a big more to understand why it's failing. This should give you a clue to do the same.